Doc/library/tkinter.rst

```diff
@@ -11,100 +11,267 @@
-      Extensive tutorial plus friendlier widget pages for some of the widgets.
-
-   `Tkinter reference: a GUI for Python <https://infohost.nmt.edu/tcc/help/pubs/tkinter/web/index.html>`_
-      On-line reference material.
-
-   `Tkinter docs from effbot <http://effbot.org/tkinterbook/>`_
-      Online reference for tkinter supported by effbot.org.
-
-   `Programming Python <http://learning-python.com/about-pp4e.html>`_
-      Book by Mark Lutz, has excellent coverage of Tkinter.
-
-   `Modern Tkinter for Busy Python Developers <https://www.amazon.com/Modern-Tkinter-Python-Developers-ebook/dp/B0071QDNLO/>`_
-      Book by Mark Rozerman about building attractive and modern graphical user interfaces with Python and Tkinter.
-
-   `Python and Tkinter Programming <https://www.manning.com/books/python-and-tkinter-programming>`_
-      Book by John Grayson (ISBN 1-884777-81-3).
-
-   Tcl/Tk documentation:
+Tkinter supports a range of Tcl/Tk versions, built either with or
+without thread support. The official Python binary release bundles Tcl/Tk 8.6
+threaded. See the source code for the :mode:`_tkinter` module
+for more information about supported versions.
+
+Tkinter is not a thin wrapper, but adds a fair amount of its own logic to
+make the experience more pythonic. This documentation will concentrate on these
+additions and changes, and refer to the official Tcl/Tk documentation for
+descriptions of underlying features.
+
+
+Architecture
+------------
+
+Tkinter provides a unified interface to several separate and distinct libraries.
+
+Tcl
+   Like Python, Tcl is a dynamic interpreted programming language. It is most
+   commonly used by C applications to embed a scripting language, or as an
+   interface to the Tk toolkit. The Tcl library has a C interface to
+   create and manage one or more instances of a Tcl interpreter, run Tcl
+   commands and scripts in those instances, and add custom commands
+   implemented in either Tcl or C. Tcl provides for an event queue, used
+   for I/O, timers, and by modules.
+   
+Tk
+   Tk is a module that can be loaded into a Tcl interpreter instance. It adds
+   Tcl commands (implemented in C) to create and manipulate GUI widgets. Each
+   :class:`Tk` object embeds its own Tcl interpreter instance with Tk loaded into
+   it. Tk's widgets are very customizable, though at the cost of a dated appearance.
+   Tk uses Tcl's event queue to generate and process GUI events. Note
+   that unlike some GUI libraries, each interpreter uses only a single thread,
+   which has implications for :mod:`tkinter` users (see `Threading model`_).
+
+Ttk
+   Themed Tk (Ttk) is a newer family of Tk widgets that provide a much better
+   appearance on different platforms than many of the classic Tk widgets.
+   Ttk is distributed as part of Tk, starting with Tk version 8.5. Python
+   bindings are provided in a separate module, :mod:`tkinter.ttk`.
+
+Tix
+   `Tix`<https://core.tcl.tk/jenglish/gutter/packages/tix.html>_ is an older
```

Comment:
Accurate.  Tix is abandoned (the last release was over 10 years ago) and deprecated, and will be removed as soon as we have any trouble building it for the Windows releases.

Frankly, I'd rather it out altogether here so we have fewer places to fix when we do remove it.

Reply:
Older is factually true. Python people are keeping it on life support but that's the extent of it. But as long as its noted as deprecated, fine.

Reply:
[As Joel Spolsky writes](https://www.joelonsoftware.com/2000/04/06/things-you-should-never-do-part-i/), _code doesn't rust._ So "older" and "abandoned" are moot points. 

Conversely, "deprecated in favor of Ttk" is the precise formulation.

Reply:
From http://tix.sourceforge.net/dist/current/man/html/contents.htm and https://www.tcl.tk/man/tcl8.5/TkCmd/contents.htm , it's not clear if all "useful" Tix widgets have Ttk replacements.

Reply:
> As Joel Spolsky writes, code doesn't rust. So "older" and "abandoned" are moot points.

I mean, it can't be deprecated _just_ because it's "old" or the original author lost interest (PyXML is a recent counterexample). It must be somehow inferior to other things to lose competition to them.

Anyway, https://github.com/roseman/cpython/pull/9